src/Test/WebServerSubscriber.php

```diff
@@ -11,49 +11,51 @@
 
 namespace FOS\HttpCache\Test;
 
-/**
- * This fake trait is used to have the same code and behavior between WebServerListener and its legacy version.
- */
-class WebServerListenerTrait
+use PHPUnit\Event\TestRunner\ExecutionStarted;
+use PHPUnit\Event\TestRunner\ExecutionStartedSubscriber;
+
+class WebServerSubscriber implements ExecutionStartedSubscriber
 {
     /**
      * PHP web server PID.
-     *
-     * @var int
      */
-    protected $pid;
+    private int $pid;
 
-    /**
-     * Make sure the PHP built-in web server is running for tests with group
-     * 'webserver'.
-     */
-    public function startTestSuite($suite)
+    public function notify(ExecutionStarted $event): void
     {
-        // Only run on PHP >= 5.4 as PHP below that and HHVM don't have a
-        // built-in web server
-        if (defined('HHVM_VERSION')) {
-            return;
-        }
-
-        if (null !== $this->pid || !in_array('webserver', $suite->getGroups())) {
```

Comment:
we used to check the groups of the test suite, but that seems not available with phpunit 10. is there a way to know which groups a test suite includes?